provisioner/aws.go

```diff
@@ -715,7 +725,29 @@ func isDoesNotExistsErr(err error) bool {
 // describes a failure because of wrong Cloudformation stack status.
 func isWrongStackStatusErr(err error) bool {
 	if awsErr, ok := err.(awserr.Error); ok {
-		if awsErr.Code() == "ValidationError" && strings.Contains(awsErr.Message(), "cannot be deleted while in status") {
+		if awsErr.Code() == cloudformationValidationErr && strings.Contains(awsErr.Message(), "cannot be deleted while in status") {
+			return true
+		}
+	}
+	return false
+}
+
+// isStackNoUpdateNeededErr returns true if the error is of type awserr.Error and
+// describes a failure because the stack template has no changes.
+func isStackNoUpdateNeededErr(err error) bool {
+	if awsErr, ok := err.(awserr.Error); ok {
+		if awsErr.Code() == cloudformationValidationErr && awsErr.Message() == cloudformationNoUpdateMsg {
+			return true
+		}
+	}
+	return false
+}
+
+// isStackUpdateInProgressErr returns true if the error is of type awserr.Error and
+// describes a failure because the stack is currently updating.
+func isStackUpdateInProgressErr(err error) bool {
+	if awsErr, ok := err.(awserr.Error); ok {
+		if awsErr.Code() == cloudformationValidationErr && (strings.Contains(awsErr.Message(), "UPDATE_IN_PROGRESS") || strings.Contains(awsErr.Message(), "UPDATE_COMPLETE_CLEANUP_IN_PROGRESS")) {
```

Comment:
nit: you can use the constants for e.g. `UPDATE_IN_PROGRESS` from here: https://github.com/aws/aws-sdk-go/blob/main/service/cloudformation/api.go#L23590-L23656

Reply:
Fixed 👍